### PR TITLE
Add basic navigation functionalities

### PR DESCRIPTION
**Pull request by Tan-Jia-Rong**
Planning to update some test cases after TypicalAddressBook is merged.

### `GoToCommand`

- [x]  Can only display 1 module at a time
- [x] Input by user is case-insensitive
- [x] Displays an empty list when module selected does not exist.

#### Possible Future work:
Cohesion with contact for v1.3
Consider other implementations of how an empty list should be handled

### `HomeCommand`

- [x] Returns unfiltered Module and Person list

#### Issues faced that needs to be discussed and fixed by v1.3
1. After filtering module list and person list, adding a module displays all the modules in the module list (unfiltered list) while person list remains unchanged
2. After filtering module list and person list, adding a person displays every person in the person list (unfiltered list) while module list remains unchanged
3. Adding tasks or links displays all the modules in the module list
4. After filtering module, couldn't add task and links to modules in unfiltered list

####Possible Solution: 
**For 1 and 2**
- Removal of `updateFilteredPersonList` and `updateFilteredModuleList` add module/person

**For 3**
- Alternative implementation of `updateFilteredModuleList` 

**For 4**
- Alternative implementation of `updateFilteredModuleList`

As we are approaching the deadline, all these implementation would be pushed back to the next mini milestone. Would like to hear your thoughts before implementing the solution proposed! 😄

### CODE REVIEW

**Comment by cheeheng** [comment]
You could just mark as draft, then nobody can enable auto-merge.

**Comment by Tan-Jia-Rong** [comment]
Yes indeed I should have uploaded it as a draft... AHAHAHA I've totally forgotten that GitHub provided such functionalities 😛

File: src/test/java/seedu/address/model/module/ModuleCodeMatchesKeywordsPredicateTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        ModuleCodeMatchesKeywordPredicate predicate = new ModuleCodeMatchesKeywordPredicate(MODULECODESTR_CS2106);
+        assertTrue(predicate.test(new ModuleBuilder().withModuleCode(MODULECODESTR_CS2106).build()));
+
+        // Mixed-case keywords
+        predicate = new ModuleCodeMatchesKeywordPredicate("cS2106");
+        assertTrue(predicate.test(new ModuleBuilder().withModuleCode(MODULECODESTR_CS2106).build()));
+    }
+
+    @Test
+    public void test_moduleCodeDoesNotMatchesKeywords_returnsFalse() {
```
Comment (teoyuqi) [inline src/test/java/seedu/address/model/module/ModuleCodeMatchesKeywordsPredicateTest.java:51]:
```suggestion
    public void test_moduleCodeDoesNotMatchKeywords_returnsFalse() {
```
There seems to be a little typo here.

File: src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private GoToCommandParser parser = new GoToCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, GoToCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFindCommand() {
+        // no leading and trailing whitespaces
```
Comment (teoyuqi) [inline src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java:25]: Perhaps for future iterations, we could have some tests for the following edge cases as well:
* inputs with leading and trailing whitespace
* inputs with invalid module code
* inputs with multiple module code listed
* input with invalid prefixes given (e.g. giving module code and phone number)

**Review by teoyuqi — COMMENTED** [review]
Looks good to me overall! Thanks for being able to work on such a tight schedule 😭

File: src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java (hunk 2 above)
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java]: Great idea! Ive implemented test cases for 1 and 2

For 3 currently, I do not feel the need to as current GoToCommand test cases only check whether the command by itself is valid

bullet point 3 shall be implemented under ModuleCodeMatchesKeywordsTest in next iteration!

**Review by teoyuqi — DISMISSED** [review]
LGTM

File: src/test/java/seedu/address/testutil/TypicalIndexes.java [hunk 3]
```diff
@@ -9,4 +9,7 @@ public class TypicalIndexes {
     public static final Index INDEX_FIRST_PERSON = Index.fromOneBased(1);
     public static final Index INDEX_SECOND_PERSON = Index.fromOneBased(2);
     public static final Index INDEX_THIRD_PERSON = Index.fromOneBased(3);
+    public static final Index INDEX_FIRST_MODULE = Index.fromOneBased(1);
```
Comment (cheeheng) [inline src/test/java/seedu/address/testutil/TypicalIndexes.java:12]: A design consideration for v1.3, maybe the index variables shouldn’t be differentiated between person and module?

**Review by cheeheng — COMMENTED** [review]
LGTM apart from these minor nitpicks.

File: src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package seedu.address.model.module;
+
+import java.util.function.Predicate;
+
+
+
+/**
+ * Tests that a {@code Module}'s {@code ModuleCode} matches exactly with the keywords given (case-insensitive).
```
[the remark this answers is not in the document]
Comment (Tan-Jia-Rong) [reply src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java:6]: What are your thought on this Im fine with both implementation. But I lean more towards having it being case-insensitive as it makes more sense.

Consider the scenario where a student mistype when creating a new module cs2103T vs CS2103T
he would have 2 copies of the same module storing different task when it is likely that the two modules refers to the same thing

A side note regarding ModuleCode as well I noticed that it allows user to type `add-module m/CS21  03T` and this would still work returning a module with modulecode CS21  03T. I feel that this should be flagged as an error in our next iteration as well

File: src/test/java/seedu/address/logic/commands/GoToCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.module.ModuleCodeMatchesKeywordPredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code GoToCommand}.
+ */
+public class GoToCommandTest {
+    private static final String MODULECODESTR_CS2106 = CS2106.getModuleCodeAsUpperCaseString();
```
[the remark this answers is not in the document]
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/logic/commands/GoToCommandTest.java:27]: Noted! Good idea I think it looks neater as well why didn't I thought of that!

File: src/test/java/seedu/address/logic/commands/GoToCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    @Test
+    public void execute_validKeywords_moduleFound() {
+        String expectedMessage = MESSAGE_MODULE_LISTED;
+        ModuleCodeMatchesKeywordPredicate predicate = new ModuleCodeMatchesKeywordPredicate(MODULECODESTR_CS2106);
+        GoToCommand command = new GoToCommand(predicate);
+        expectedModel.updateFilteredModuleList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(CS2106), model.getFilteredModuleList());
+    }
+}
```
[the remark this answers is not in the document]
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/logic/commands/GoToCommandTest.java:78]: Yes it is case-insensitive. I shall include the test

File: src/test/java/seedu/address/logic/commands/HomeCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test
+    public void execute_personListIsNotFiltered_showsSameList() {
+        assertCommandSuccess(new HomeCommand(), personModel, HomeCommand.MESSAGE_SUCCESS, expectedPersonModel);
+    }
+
+    @Test
+    public void execute_moduleListIsFiltered_showsEverything() {
+        // To be added with filtered model
```
[the remark this answers is not in the document]
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/logic/commands/HomeCommandTest.java]: Hmm... I do not have this comment on my current copy could be from previous iterations

File: src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import static seedu.address.testutil.TypicalModules.CS2106;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.GoToCommand;
+import seedu.address.model.module.ModuleCodeMatchesKeywordPredicate;
+
+public class GoToCommandParserTest {
+
+    private static final String MODULECODESTR_CS2106 = CS2106.getModuleCodeAsUpperCaseString();
```
[the remark this answers is not in the document]
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/logic/parser/GoToCommandParserTest.java:19]: yep ill do so

File: src/test/java/seedu/address/testutil/TypicalIndexes.java (hunk 3 above)
Comment (Tan-Jia-Rong) [reply src/test/java/seedu/address/testutil/TypicalIndexes.java:12]: Hmm then shall I change to it INDEX_FIRST, INDEX_SECOND and INDEX_THIRD? I thought of putting it in as it adds more clarity to the person writing the test code. ie, when I write INDEX_FIRST_MODULE, I know that I am taking the first index of module.

**Review by teoyuqi — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/commands/HomeCommandTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (cheeheng) [reply src/test/java/seedu/address/logic/commands/HomeCommandTest.java]: Yes, you fixed it.

File: src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (cheeheng) [reply src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java:6]: Good idea, I might consider implementing this.

**Review by cheeheng — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/testutil/TypicalIndexes.java (hunk 3 above)
Comment (cheeheng) [reply src/test/java/seedu/address/testutil/TypicalIndexes.java:12]: If putting `INDEX_FIRST_MODULE` makes it clearer, then perhaps you probably should leave it as it is?

File: src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ekweirui) [reply src/main/java/seedu/address/model/module/ModuleCodeMatchesKeywordPredicate.java:6]: Keep me in the loop regarding this issue as the current implementation of contacts is case-sensitive as well @cheeheng
